sdk/src/main/scala/io/horizen/account/state/AccountState.scala

```diff
@@ -449,7 +450,9 @@ class AccountState(
 
   override def isEoaAccount(address: Address): Boolean = using(getView)(_.isEoaAccount(address))
 
-  override def isSmartContractAccount(address: Address): Boolean = using(getView)(_.isSmartContractAccount(address))
+  override def isEvmSmartContractAccount(address: Address): Boolean = using(getView)(_.isEvmSmartContractAccount(address))
```

Comment:
Agree, the approach would be dangerous. I removed those methods and change back also the original name. The approach now is to let Eoa msg processor to consume any pre-fork transaction to the native smart contract (if any); consider that since fork dependant native smart contracts are not initialized at genesis state, their msg processor must be placed before the Eoa msg processor.